src/test/java/com/exasol/spark/s3/S3CleanupIT.java

```diff
@@ -0,0 +1,239 @@
+package com.exasol.spark.s3;
```

Comment:
Please (optionally) see proposed refactoring in [branch feature/#159-cleanup-ck-proposal](https://github.com/exasol/spark-connector/blob/feature/%23159-cleanup-ck-proposal/src/test/java/com/exasol/spark/s3/S3CleanupIT.java)

Reply:
Ohh, unfortunately, I have to revert these changes, I forgot that each map or flatMap transformations are serialized to the threads, and our FailMax class also serialized as it is, which does not keep global count among the threads.

Let's us address this again, maybe we could refactor it differently.

Reply:
Hm, I somehow feared that